src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -813,26 +813,25 @@ static inline JSC::EncodedJSValue jsBufferConstructorFunction_concatBody(JSC::JS
     }
 
     for (unsigned i = 0; i < arrayLength; i++) {
-        auto element = array->getIndex(lexicalGlobalObject, i);
+        JSValue element = array->getIndex(lexicalGlobalObject, i);
         RETURN_IF_EXCEPTION(throwScope, {});
 
-        auto* typedArray = JSC::jsDynamicCast<JSC::JSUint8Array*>(element);
-        if (!typedArray) {
-            throwTypeError(lexicalGlobalObject, throwScope, "Buffer.concat expects Uint8Array"_s);
-            return {};
-        }
-
-        if (UNLIKELY(typedArray->isDetached())) {
-            throwVMTypeError(lexicalGlobalObject, throwScope, "Uint8Array is detached"_s);
-            return {};
-        }
+        if (auto* bufferView = JSC::jsDynamicCast<JSC::JSArrayBufferView*>(element)) {
+            if (UNLIKELY(bufferView->isDetached())) {
+                throwVMTypeError(lexicalGlobalObject, throwScope, "ArrayBufferView is detached"_s);
+                return {};
+            }
 
-        auto length = typedArray->length();
+            auto length = bufferView->byteLength();
 
-        if (length > 0)
-            args.append(element);
+            if (length > 0)
+                args.append(element);
 
-        byteLength += length;
+            byteLength += length;
+        } else {
+            throwTypeError(lexicalGlobalObject, throwScope, "Buffer.concat expects Buffer or Uint8Array"_s);
```

Comment:
This throwTypeError -- is that throwing the error with the code property that is expected or is it not doing that?